api/controllers/PostController.js

```diff
@@ -392,6 +400,36 @@ module.exports = {
       }
     })
     .then(() => res.ok({}), res.serverError)
-  }
+  },
+
+  checkFreshnessForCommunity: function (req, res) {
+    var foundPosts
+    var resolve = value => foundPosts = value
```

Comment:
oh, right. oops. well, then you'll want to refactor `findPosts` to separate the query logic from the presentation logic, and use only the former for these routes.